### DISCUSSION TITLE
User login/register journeys

### DISCUSSION THREAD

**Discussion by squigglybob** [Pray4Movement/prayer-global-porch#37] [Ideas]
# User  Login/Register Journey

@prykon @ChrisChasm @corsacca @himayrunner @ahillbilly

The login/registration screen will look very like the Zume one.

## Overarching questions

1. Where to put the login button?
2. Where to put the register button?
3. When to prompt to login?
4. When to prompt to register?
5. Can the app save the login state? And for how long?
6. Do we promote the logged in experience over the anonymous one?
7. How do the changes made affect how the user feels if they want to stay anonymous?

I think that registering after your experience would be best, or at least promoting it at that point, with the carrot of personalised map and stats, and creating challenges.

So on click done, could offer login/register buttons.

## Once you have registered:

a) It should either keep you logged in.
b) Allow you to login easily from the start.
c) Remind you once done if you weren't logged in (same as previous)

## analytics

Do we have stats for how many prayer warriors actually hit the Done screen?
This would help us better decide if that is the best place to catch people or not for registering/logging in.

## Current user journey

```mermaid
journey
	title Going to pray
	section Entering App
		Click start praying: 5
	section Praying
		Pray + repeat: 5
		View stats: 1
```
I think that the current user journey at least for praying on the global map is pretty streamlined and simple.

Keeping that simplicity while introducing the login/registration options is key.

## Possible Login/register user journeys

Not sure how helpful these images are in this discussion, but I'm playing with these things to get a feel for how useful they are/aren't

```mermaid
journey
	title Register first
	section Entering App
		Find Register button: 3
	section Praying
		Pray and repeat: 7
		View stats: 7
```
I think you should be able to register at the start, and it should be easy to find.

```mermaid
journey
	title Logging in first
	section Entering App
		Already logged in?: 7
		Find login button?: 3
		Click pray: 5
	section Praying
		Pray and repeat: 7
		View stats: 7
```
Logging in at the start should also be easy to find and do.

```mermaid
journey
	title Register/login on Done
	section Enter app
		click start praying: 7
	section Pray
		pray and repeat: 7
		want to save stats?: 5
		login/register: 5
		View stats: 7
```
As long as people reliably click on the Done button when they are done praying, the Done screen could be a good place to promote the registering (if they aren't already logged on) 

## Possible places for the login/signup buttons

On the home screen we could put them in the top right on the nav bar where the "start praying" button currently is, and lean more on the big "start praying" call to action button in the middle of the map in the center of the screen.

Then the top bar would look a bit like this
![image](https://user-images.githubusercontent.com/9816214/201667137-f8116623-1404-4c31-a9ec-7e820cb021d8.png)

**Comment by corsacca** [comment]
@squigglybob this page is awesome, I love the graphics!

I think, the Login/Register button should be easily accessible.
I think we should keep the current experience without login. 
I think we should ask them to login on the celebrate page. Want to save stats, see where you have prayed, join groups and experience custom prayer paths? Login.

If we can, having the login not expire would be awesome. We aren't dealing with sensitive data here.

```mermaid
journey
	title Register/login on Done
	section Enter app
		click start praying: 7
	section Pray
		pray and repeat: 7
		want to save stats?: 5
		login/register: 5
		Keep Praying: 7
		View stats button: 5
```

**Reply by squigglybob** [reply]
> @squigglybob this page is awesome, I love the graphics!
> 
> I think, the Login/Register button should be easily accessible.

👍🏻 

> I think we should keep the current experience without login.

👍🏻 👍🏻 

> I think we should ask them to login on the celebrate page. Want to save stats, see where you have prayed, join groups and experience custom prayer paths? Login.

It could go there, but the downside to this is that it could detract from the prayer experience of going from one to the next and pull them out of the prayer cycle.

> If we can, having the login not expire would be awesome. We aren't dealing with sensitive data here.

👍🏻

**Reply by squigglybob** [reply]
looking at your possible user journey, if there were a way of making the login/register super smooth and quick and not janky at all, it could slot in as a, "hey, want to make sure your stats are saved? login/register" but with a keep praying button to keep them on the praying journey.

The Done button could become a "Done + stats" kinda thing too, which could start to take more people to the Done screen if not as many are getting there at the moment ( which we would need analytics to check )

**Comment by himayrunner** [comment]
@squigglybob I like the login/signup buttons' location. 

I'm a little confused on the smiling faces -- this is not a mockup, right? this is just a cool way of exploring a user journey in Github, right? :)

**Reply by squigglybob** [reply]
@himayrunner yes I think the purpose of the images is to show some sort of scale of how each step in the journey feels to a user. I have just put  in what I think each step does/could feel like, and put a :| smiley for the steps that depend on how/where we put buttons

**Comment by prykon** [comment]
A big plus could be to keep prayer data stored client-side for non-registered users and then add that to their prayer profile when they register.

**Reply by squigglybob** [reply]
totally
